crates/typst-utils/src/format.rs

```diff
@@ -0,0 +1,135 @@
+//! Different formatting functions.
+
+use std::time::Duration;
+
+/// Returns value with `n` digits after floating point where `n` is `precision`.
+/// Standard rounding rule applies (if `n+1`th digit >= 5 then round up).
+/// If `value` is +/- infinity returns `value`.
+///
+/// Will panic for a very large `value` and `precision` (if the `f64` limit is hit).
+/// In practice, it should never panic, otherwise the implementation have to be
+/// more robust.
+pub fn round_with_precision(value: f64, precision: u8) -> f64 {
+    let offset = 10_f64.powi(precision.into());
+    if value.is_infinite() {
+        return value;
+    }
```

Comment:
```suggestion
    if value.is_infinite() {
        return value;
    }
    let offset = 10_f64.powi(precision.into());
```

Reply:
No fair! I thought of this first, but haven't fixed it because I have to go to sleep. I'll fix it.